src/containers/course-sections-list/CourseSectionsList.tsx

```diff
@@ -45,7 +50,30 @@ const CourseSectionsList: FC<CourseSectionsListProps> = ({
   addNewSection
 }) => {
   const { enabled } = useDroppable()
-  const { openMenu, renderMenu, closeMenu, anchorEl } = useMenu()
+  const [active, setActive] = useState<Active | null>(null)
+  const activeItem = useMemo(
+    () => items.find((item) => item.id === active?.id),
+    [active, items]
+  )
+
+  const sensors = useSensors(useSensor(PointerSensor), useSensor(TouchSensor))
+
+  const onDragEnd = useCallback(
+    (event: DragEndEvent) => {
+      const { active, over } = event
+
+      if (active && over) {
+        const activeIndex = items.findIndex((item) => item.id === active.id)
+        const overIndex = items.findIndex((item) => item.id === over.id)
+        if (activeIndex !== -1 && overIndex !== -1) {
+          setSectionsItems(arrayMove(items, activeIndex, overIndex))
+        }
```

Comment:
I have seen this code before. Maybe we can create custom hook something like `useDndSensor` and extract repeated logic (`useSensors`, `onDrag` etc)

Reply:
Yes, it's good idea. Also we have a problem with different id key, so i created generic and now it's works

Reply:
Good for you!